src/test/java/io/appform/jsonrules/PreOperationTest.java

```diff
@@ -0,0 +1,1368 @@
+                .evaluate(context));
+        Assert.assertTrue(EqualsExpression.builder()
+                .path("/epochTime")
+                .preoperation(EpochOperation.builder().operand("week_of_month").zoneOffSet("+05:30").build())
+                .value(dateTime.atOffset(ZoneOffset.of("+05:30")).get(ChronoField.ALIGNED_WEEK_OF_MONTH))
+                .build()
+                .evaluate(context));
+        Assert.assertTrue(EqualsExpression.builder()
+                .path("/epochTime")
+                .preoperation(EpochOperation.builder().operand("week_of_year").zoneOffSet("+05:30").build())
+                .value(dateTime.atOffset(ZoneOffset.of("+05:30")).get(ChronoField.ALIGNED_WEEK_OF_YEAR))
+                .build()
+                .evaluate(context));
+        Assert.assertTrue(EqualsExpression.builder()
+                .path("/epochTime")
+                .preoperation(EpochOperation.builder().operand("month_of_year").zoneOffSet("+05:30").build())
+                .value(dateTime.atOffset(ZoneOffset.of("+05:30")).get(ChronoField.MONTH_OF_YEAR))
+                .build()
+                .evaluate(context));
+        
+        // Would only match with the specified ZoneOffSet
+        Assert.assertFalse(EqualsExpression.builder()
+                .path("/epochTime")
+                .preoperation(EpochOperation.builder().operand("minute_of_hour").zoneOffSet("+05:30").build())
+                .value(dateTime.atOffset(ZoneOffset.UTC).get(ChronoField.MINUTE_OF_HOUR))
+                .build()
+                .evaluate(context));
+        Assert.assertFalse(EqualsExpression.builder()
+                .path("/epochTime")
+                .preoperation(EpochOperation.builder().operand("hour_of_day").zoneOffSet("+05:30").build())
+                .value(dateTime.atOffset(ZoneOffset.UTC).get(ChronoField.HOUR_OF_DAY))
+                .build()
+                .evaluate(context));
+        
+        // Default ZoneOffSet considered is UTC
+        Assert.assertTrue(EqualsExpression.builder()
+                .path("/epochTime")
+                .preoperation(EpochOperation.builder().operand("minute_of_hour").build())
+                .value(dateTime.atOffset(ZoneOffset.UTC).get(ChronoField.MINUTE_OF_HOUR))
+                .build()
+                .evaluate(context));
+        Assert.assertTrue(EqualsExpression.builder()
+                .path("/epochTime")
+                .preoperation(EpochOperation.builder().operand("hour_of_day").build())
+                .value(dateTime.atOffset(ZoneOffset.UTC).get(ChronoField.HOUR_OF_DAY))
+                .build()
+                .evaluate(context));
+        
+        Assert.assertFalse(EqualsExpression.builder()
+        		.path("/dateTime")
+        		.preoperation(EpochOperation.builder().operand("minute_of_hour").build())
+        		.value(dateTime.atOffset(ZoneOffset.UTC).get(ChronoField.MINUTE_OF_HOUR))
+        		.build()
+        		.evaluate(context));
+	}
+
```

Comment:
Add tests that use the serialized version of the ops to check serialization/deserialization issues.